### PR TITLE
Initial jobs with dagger+github actions

### PR DESCRIPTION
**Pull request by roidelapluie**
Signed-off-by: Julien Pivotto <roidelapluie@o11y.eu>

### CODE REVIEW

**Comment by Nexucis** [comment]
looks good overall, maybe you would like to introduce a way to format the cue files.

the binary `cue` provides you a way to do it. Can be used like that for example: 

`cue fmt .dagger/dagger.cue`

File: .dagger/dagger.cue [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package main
+
+import (
+	"dagger.io/dagger"
+	"dagger.io/dagger/core"
+	"github.com/prometheus/promci"
```
Comment (Nexucis) [inline .dagger/dagger.cue:6]: I would apply the same logic than in go to sort the import here.

all `dagger` import should come first. Then you have an empty line, and then you have the rest.

File: .github/workflows/promci.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    steps:
+      -
+        name: Checkout
+        uses: actions/checkout@v3
+      -
+        name: Dagger
+        uses: dagger/dagger-for-github@v3
+        with:
+          workdir: .dagger
+          cmds: do ui
```
Comment (Nexucis) [inline .github/workflows/promci.yml:31]: I guess you can already removed these jobs from circle-ci.

**Comment by Nexucis** [comment]
don't you want to add the `build` job too ?

File: .github/workflows/promci.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+name: dagger
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
```
Comment (LeviHarrison) [inline .github/workflows/promci.yml:9]: Are we still doing windows tests?

```suggestion
    strategy:
      matrix:
        os: [ubuntu-latest, windows-latest]
    runs-on: ${{ matrix.os }}
```

**Review by LeviHarrison — COMMENTED** [review]
I assume there's no way not to have a vendor directory for dagger dependencies?

**Comment by Nexucis** [comment]
this PR can be closed in favor of #11235

**Pull request closed without merging**